src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -75,33 +76,207 @@ QIcon icon_ ## FUNC( Gui::BitmapFactory().pixmap(ICONSTR) ); \
+                    }));
+
+        icons.emplace(std::piecewise_construct,
+                      std::forward_as_tuple(Base::Type::badType()),
+                      std::forward_as_tuple(std::initializer_list<std::pair<const Sketcher::PointPos, std::tuple<QIcon,QIcon,QIcon,QIcon>>> {
+                        { Sketcher::PointPos::none , getMultIcon("Sketcher_Element_SelectionTypeInvalid") },
+                    }));
+
+    }
+
+    const QIcon & getIconImpl(Base::Type type, Sketcher::PointPos pos, ElementWidgetIcons::Type icontype) {
+
+        auto typekey = icons.find(type);
+
+        if( typekey == icons.end()) { // Not supported Geometry Type - Defaults to invalid icon
+            typekey = icons.find(Base::Type::badType());
+            pos = Sketcher::PointPos::none;
+        }
+
+        auto poskey = typekey->second.find(pos);
+
+        if ( poskey == typekey->second.end()) { // invalid PointPos for type - Provide Invalid icon
+            typekey = icons.find(Base::Type::badType());
+            pos = Sketcher::PointPos::none;
+            poskey = typekey->second.find(pos);
+        }
+
+        if (icontype == Type::Normal)
+            return std::get<0>(poskey->second);
+        else if (icontype == Type::Construction)
+            return std::get<1>(poskey->second);
+        else if (icontype == Type::External)
+            return std::get<2>(poskey->second);
+        else //internal
+            return std::get<3>(poskey->second);
+
+        // We should never arrive here, as badtype, PointPos::none must exist.
+        throw Base::ValueError("Icon for Invalid is missing!!");
+
+    }
+
+    std::tuple<QIcon, QIcon, QIcon, QIcon> getMultIcon(const char* name)
+    {
+        int hue, sat, val, alp;
+        QIcon Normal = Gui::BitmapFactory().iconFromTheme(name);
+        QImage imgConstr(Normal.pixmap(qAsConst(Normal).availableSizes()[0]).toImage());
+        QImage imgExt(imgConstr);
+        QImage imgInt(imgConstr);
+
+        //Create construction/external/internal icons by changing colors.
+        for(int ix=0 ; ix<imgConstr.width() ; ix++) {
+            for(int iy=0 ; iy<imgConstr.height() ; iy++) {
+                QColor clr(imgConstr.pixelColor(ix,iy));
+                clr.getHsv(&hue, &sat, &val, &alp);
+                if (alp > 127 && hue >= 0) {
+                    if (sat > 127 && (hue > 330 || hue < 30)) { //change the color of red points.
+                        //clr.setHsv((hue + 240) % 360, sat, val, alp); //Construction lines have red points !
+                        //imgConstr.setPixelColor(ix, iy, clr);
+                        //clr.setHsv((hue + 300) % 360, sat, val, alp); //External lines have red points !
+                        //imgExt.setPixelColor(ix, iy, clr);
+                        clr.setHsv((hue + 60) % 360, (int) sat / 3, (int) val * 8 / 7, alp); //Internal points have same color has internal edges.

```

Comment:
`sat / 3` will be IMO a bit close to white and possibly hard to distinguish based on your monitor settings. Maybe `sat / 2` would be enough.
Also be careful what casting applies to. Better use parenthesis.
Also be careful that `val * 8 / 7` can overflow. I wouldn't touch this.
At the end :+1: 
```suggestion
                        clr.setHsv((hue + 60) % 360, (int) (sat / 2), val * 8 / 7, alp); //Internal points have same color has internal edges.

```

Reply:
I used a paint program to compare the colors and calculate the required values to match the geos on the view by screenshoting. The geo on the view had a sat of 28. val was around 90. Hence the 1/3.
Now comparing on paint the values of pixel colors, they are very close.
So I think it should not be dependant on screen parameters. If the user's screen show the icon white, it should show the geos white too.

Reply:
Do whatever you want. This will broken as soon as user changes default color. So IMO better keep code simple.

Reply:
As a feedback, I can say that in my computer the internal alignment color in icons is much lighter than the 3D view colour. I am not good at differentiating colours, I see it very light yellow, difficult to differentiate from white for edges.

Reply:
I'm fine with modifying the value. I did my best to match the color but if you can do better it's great.

Reply:
@abdullahtahiriyo Is that so important ? The blue isn't correct probably, as well as the purple. ;) Back in the days, the idea was just to have colors that can easily "distinguish" geometry types in the element list. ;)